pyaptamer/datasets/_loaders/load_aptamer_interactions.py

```diff
@@ -0,0 +1,235 @@
+    """Normalize interaction present column in the dataset.
+
+    This is a private helper function for data preprocessing.
+    Currently a placeholder for future implementation.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The dataframe to normalize
+
+    Notes
+    -----
+    This function is currently not implemented and serves as a placeholder
+    for future data normalization functionality.
+    """
+    # TODO: Implement interaction present normalization
+    return
+
+
+def load_aptamer_interactions(
+    path: str | Path,
+    *,
+    encoding: str | None = None,
+    **read_csv_kwargs,
+) -> pd.DataFrame:
+    """Load an aptamer interactions CSV file into a pandas DataFrame.
+
+    This function provides robust CSV loading with automatic encoding detection
+    and error handling for various file formats commonly found in biological
+    datasets.
+
+    Parameters
+    ----------
+    path : str or Path
+        Path to the CSV file containing aptamer interaction data
+    encoding : str, optional
+        Specific file encoding to use. If None (default), multiple common
+        encodings will be tried automatically
+    **read_csv_kwargs
+        Additional keyword arguments passed directly to pandas.read_csv()
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame containing the loaded aptamer interaction data
+
+    Raises
+    ------
+    RuntimeError
+        If the CSV file cannot be read with any of the attempted encodings
+
+    Notes
+    -----
+    The function attempts the following encodings in order:
+    - utf-8
+    - utf-8-sig (for files with BOM)
+    - latin-1
+    - cp1252
+    - windows-1252
+    """
+    candidate_encodings = (
+        [
+            "utf-8",
+            "utf-8-sig",  # For files with byte order mark
+            "latin-1",
+            "cp1252",  # Common Windows encoding
+            "windows-1252",  # Alternative Windows encoding
+        ]
+        if encoding is None
+        else [encoding]
+    )
+
+    last_error: Exception | None = None
+
+    for enc in candidate_encodings:
+        try:
+            df = pd.read_csv(path, encoding=enc, **read_csv_kwargs)
+            return df
+        except Exception as e:
+            last_error = e
+            continue
+
+    raise RuntimeError(
+        f"Failed to read CSV at {path} with candidate encodings "
+        f"{candidate_encodings}: {last_error}"
+    )
+
+
+def load_interactions(
+    path: str | Path,
+    *,
+    encoding: str | None = None,
+    **read_csv_kwargs,
+) -> pd.DataFrame:
+    """Load interaction data from a CSV file.
+
+    This is a convenience alias for load_aptamer_interactions() with identical
+    functionality and parameters.
+    """
+    return load_aptamer_interactions(
+        path=path,
+        encoding=encoding,
+        **read_csv_kwargs,
+    )
+
+
+def load_aptadb(
+    dataset_name: str = "satarupadeb/aptamer-interactions",
```

Comment:
Reminder for when your PR is ready to be merged we should move this dataset to an account under gcos.